src/core/tiledmesh/qgscesiumtilesdataprovider.cpp

```diff
@@ -0,0 +1,183 @@
+/***************************************************************************
+                         qgscesiumtilesdataprovider.cpp
+                         --------------------
+    begin                : June 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgscesiumtilesdataprovider.h"
+#include "qgsproviderutils.h"
+#include "qgsapplication.h"
+#include "qgsprovidersublayerdetails.h"
+#include "qgsthreadingutils.h"
+
+#include <QUrl>
+#include <QIcon>
+
+///@cond PRIVATE
+
+#define PROVIDER_KEY QStringLiteral( "cesium" )
```

Comment:
In the off chance that cesium pushes another dataset format we'd want to embrace in the future, should we call this cesiummesh, cesiumtiles, or cesium3d instead?

Reply:
I'll go with "cesiumtiles" :+1: